sympy/liealgebras/cartan_base.py

```diff
@@ -0,0 +1,295 @@
+from sympy.matrices.dense import MutableDenseMatrix
+from sympy.core import Basic
+from sympy.matrices import zeros, Matrix, eye, ones
+from sympy.core.sympify import _sympify
+
+class Standard_Cartan(Basic):
+    """
+    Semi-Concrete base class for Cartan types such as A4, etc. In this module we make
+    certain choices about how to represent the algebra.  This class is never meant to be
+    called directly. It is exported as `CartanType` and should be called that way.
+
+    Examples
+    ========
+    >>> from sympy.liealgebras import CartanType
+    >>> CartanType("G2")
+    TypeG('G', 2)
+
+    Sources
+    =======
+    - https://inspirehep.net/literature/181168
+    - https://www.researchgate.net/publication/252487471_Group_Theory_A_Physicist's_Survey
+    - https://arxiv.org/pdf/1206.6379.pdf
+    """
+
+    def __new__(cls, series, n):
+        n= _sympify(n)
+        return super().__new__(cls, series, n)
+
+    @property
+    def rank(self):
+        """
+        Returns the rank of the Lie algebra
+        """
+        return self.args[1]
+
+    @property
+    def series(self):
+        """
+        Returns the type of the Lie algebra
+        """
+        return self.args[0]
+
+    def cartan_matrix(self):
+        r"""
+        Returns the Cartan Matrix for the algebra. For a
+        given simple Lie algebra the elements $a_{ij}$ can be
+        generated by
+
+        .. math::
+            a_{ji} = 2 \langle\alpha_i, \alpha_j\rangle / \langle\alpha_j, \alpha_j\rangle
+
+        where $a_i$ is the i'th simple root and $\langle,\rangle$ is the scalar product.
+
+        Sources
+        =======
+        - https://en.wikipedia.org/wiki/Cartan_matrix
+        - https://mathworld.wolfram.com/CartanMatrix.html
+
+        """
+        r = self.rank
+        cartan_matrix = zeros(r,r)
+        for i, sr_i in enumerate(self.simple_roots()):
+            for j, sr_j in enumerate(self.simple_roots()):
+                cartan_matrix[j,i] = 2 * sr_i.dot(sr_j) / sr_i.dot(sr_i)
+        return cartan_matrix
+
+    def cocartan_matrix(self):
+        r"""
+        Returns the Cocartan matrix. The Cocartan matrix
+        row elements, $c_i$, are built from covectors of simple roots, $\alpha_i$.
+
+        .. math::
+            c_i = 2 \alpha_i / \langle \alpha_i, \alpha_i \rangle
+
+        This relationship is helpful in defining the orthogonal base where
+        multiplying by some weight, $w_j$, yields the unit matrix.
+
+        .. math::
+            \langle c_i, w_j \rangle = \delta_{ij}
+
+        """
+        return Matrix([
+                2 * x / x.dot(x) for x in self.simple_roots()])
+
+    def omega_matrix(self):
+        """
+        Returns the omega matrix, which rows are built from
```

Comment:
I think that this should be something like

    Returns the omega matrix whose rows are built from